aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move

```diff
@@ -51,120 +52,171 @@ module 0x1::aggregator_v2_test {
         assert!(aggregator_v2::read_snapshot(&snapshot2) == std::string::utf8(b"before42after"), 0);
     }
 
+    fun insert<Element: store>(account: &signer, i: u64, e: Element) acquires AggregatorStore {
+        let addr = signer::address_of(account);
+        let data = &mut borrow_global_mut<AggregatorStore<Element>>(addr).data;
+        let outer = i / 10;
+        let inner = i % 10;
+        if (!table::contains(data, outer)) {
+            table::add(data, outer, vector::empty());
+        };
 
-    /// Checks that the ith aggregator has expected value. Useful to inject into
-    /// transaction block to verify successful and correct execution.
-    public entry fun check(account: &signer, i: u64, expected: u128) acquires AggregatorStore {
+        let vector_data = table::borrow_mut(data, outer);
+        if (vector::length(vector_data) == inner) {
+            vector::push_back(vector_data, option::none());
```

Comment:
Not entirely sure the motivation behind this way of organizing the data in AggregatorStore. 
Why not create an aggregator/snapshot and insert at ith index of the data?